### PR TITLE
[#11202] Instructor checking activity logs: start time is not used correctly

### LINKED ISSUE #11202: Instructor checking activity logs: start time is not used correctly
page: `/web/instructor/logs`

image

Looks like 23.59 is interpreted as 'start of the day' when it should be 'end of the day'. 
As this is a 'search' feature, a mistake here has serious consequences because the instructor can be mislead into thinking that a student didn't access the submission page when in fact s/he did.

### PR DESCRIPTION
**Pull request by halfwhole**
Fixes #11202 

Changes the search to use the timezone of the feedback course/session, instead of the user web browser's timezone.

This assumes that the feedback course timezone is the same as that of its feedback sessions, which makes sense (I hope) as setting the timezone of a course changes those of its sessions too.

Before (browser in UTC+8, course in UTC):

Screenshot 2021-08-12 at 9 46 33 AM

After (browser in UTC+8, course in UTC):

Screenshot 2021-08-12 at 9 50 55 AM

### CODE REVIEW

**Review by t-cheepeng — APPROVED** [review]
LGTM

**Comment by halfwhole** [comment]
Sure sorry about it! I've updated the above with before and after screenshots.

**Review by moziliar — APPROVED** [review]
Yup, change LGTM!

File: src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.spec.ts [hunk 1]
```diff
@@ -243,11 +243,11 @@ describe('InstructorAuditLogsPageComponent', () => {
...
   });
 
-  it('should search for logs when search button is clicked', () => {
+  it('should search for logs using feedback course timezone when search button is clicked', () => {
     const logSpy: Spy = spyOn(logService, 'searchFeedbackSessionLog').and
         .returnValue(of({ feedbackSessionLogs: [testLogs1, testLogs2] }));
-    const timeSpy: Spy = spyOn(timezoneService, 'resolveLocalDateTime').and
-        .returnValue(0);
+    const timeSpy: Spy = spyOn(timezoneService, 'resolveLocalDateTime').and.callThrough();
+    const tzOffset: number = timezoneService.getTzOffsets()['Asia/Singapore'];
```
Comment (wkurniawan07) [inline src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.spec.ts:250]: If I understand this right, `Asia/Singapore` should not be a magic string but rather a field from `testCourse1`?

**Review by wkurniawan07 — APPROVED** [review]
LGTM; minor suggestion, can merge without it

File: src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.spec.ts (hunk 1 above)
Comment (halfwhole) [reply src/web/app/pages-instructor/instructor-audit-logs-page/instructor-audit-logs-page.component.spec.ts]: Right indeed, I've changed it to reference `testCourse1.timezone` instead

**Pull request merged**